storm-core/src/clj/backtype/storm/cluster.clj

```diff
@@ -54,6 +54,10 @@
     (when (Utils/isZkAuthenticationConfiguredTopology topo-conf)
       [(first ZooDefs$Ids/CREATOR_ALL_ACL)
        (ACL. ZooDefs$Perms/READ (Id. "digest" (DigestAuthenticationProvider/generateDigest payload)))])))
+ 
+(defn is-nimbus?
```

Comment:
How about removing this and let the caller of mk-distributed-cluster-state decides whether writer and reader should be separated? (I mean additional argument or multimethod.)

I think it can achieve loose coupling since ClusterState doesn't need to know current daemon is nimbus or not. It just wants to know whether writer and reader should be separated or not.